mypyc/genops.py

```diff
@@ -3395,15 +3399,91 @@ def visit_yield_expr(self, expr: YieldExpr) -> Value:
         next_block = BasicBlock()
         next_label = len(cls.blocks)
         cls.blocks.append(next_block)
-        self.assign(cls.next_label_target, self.add(LoadInt(next_label)), expr.line)
+        self.assign(cls.next_label_target, self.add(LoadInt(next_label)), line)
         self.add(Return(retval))
         self.activate_block(next_block)
 
-        self.add_raise_exception_blocks_to_generator_class(expr.line)
+        self.add_raise_exception_blocks_to_generator_class(line)
 
         assert cls.send_arg_reg is not None
         return cls.send_arg_reg
 
+    def visit_yield_from_expr(self, o: YieldFromExpr) -> Value:
+        # This is basically an implementation of the code in PEP 380.
+
+        # TODO: do we want to use the right types here?
+        result = self.alloc_temp(object_rprimitive)
+        to_yield_reg = self.alloc_temp(object_rprimitive)
+        received_reg = self.alloc_temp(object_rprimitive)
+        iter_reg = self.maybe_spill_assignable(
+            self.primitive_op(iter_op, [self.accept(o.expr)], o.line))
+
+        stop_block, main_block, done_block = BasicBlock(), BasicBlock(), BasicBlock()
+        _y_init = self.primitive_op(next_raw_op, [self.read(iter_reg)], o.line)
+        self.add(Branch(_y_init, stop_block, main_block, Branch.IS_ERROR))
+
+        # If the initial next() raised, assume it was a StopIteration and extract
```

Comment:
Here "assume" can be understood in a way that this only works if the exception was `StopIteration`, but the next statement implies that this isn't the case. Maybe reword?